Tasks/AzureFileCopyV4/AzureFileCopyRemoteJob.ps1

```diff
@@ -0,0 +1,114 @@
+$AzureFileCopyRemoteJob = {
+    param(
+        [string]$containerURL,
+        [string]$targetPath,
+        [string]$containerSasToken,
+        [string]$additionalArguments,
+        [switch]$CleanTargetBeforeCopy,
+        [switch]$EnableDetailedLogging
+    )
+
+    function Write-DetailLogs
+    {
+        [CmdletBinding()]
+        param(
+            [string]$message
+        )
+
+        if($EnableDetailedLogging)
+        {
+            Write-Verbose $message
+        }
+    }
+
+    try
+    {
+        $useDefaultArguments = ($additionalArguments -eq "")
+
+        #argument to check whether azcopy.exe needs to be downloaded on VM or it is already present on VM
+        $shouldDownload = $false
+
+        if($CleanTargetBeforeCopy)
+        {
+            if (Test-Path $targetPath -PathType Container)
+            {
+                Get-ChildItem -Path $targetPath -Recurse -Force | Remove-Item -Force -Recurse
+                Write-DetailLogs "Destination location cleaned"
+            }
+            else
+            {
+                Write-DetailLogs "Folder at path $targetPath not found for cleanup."
+            }
+        }
+
+        try
+        {
+            $azCopyVersionCommand = azcopy --version
+            $azCopyVersion = $azCopyVersionCommand.split(' ')[2]
```

Comment:
This logic can break if the version format got updated for azcopy.

Reply:
Resolved offline